app/src/control_msg.c

```diff
@@ -115,12 +115,15 @@ sc_control_msg_serialize(const struct sc_control_msg *msg, unsigned char *buf) {
             return 28;
         case SC_CONTROL_MSG_TYPE_INJECT_SCROLL_EVENT:
             write_position(&buf[1], &msg->inject_scroll_event.position);
-            sc_write32be(&buf[13],
-                             (uint32_t) msg->inject_scroll_event.hscroll);
-            sc_write32be(&buf[17],
-                             (uint32_t) msg->inject_scroll_event.vscroll);
-            sc_write32be(&buf[21], msg->inject_scroll_event.buttons);
-            return 25;
+            // map [-1, 1] to [0, 1], then to uint16
+            uint16_t hscroll =
+                to_fixed_point_16((msg->inject_scroll_event.hscroll + 1.0f) / 2.0f);
```

Comment:
I don't know what's the best way to do this.

Reply:
This looks fine to me.